index.js

```diff
@@ -10,8 +10,51 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+// retrieving colors data using fetch:
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  // fetch data from API:
+  const response = await fetch(COLORS);
+
+  if(!response.ok) {
+    throw Error('Error: Looks like something went wrong. Failed to fetch colors.');
+  }
+
+  const colors = await response.json();
+
+  let filteredColors = colors;
+
+  // filter by name
+  if(name) {
+    filteredColors = filteredColors.filter(function (color) {
+      return color.name.toLowerCase().includes(name.toLowerCase());
```

Comment:
Remember to DRY your code. The string comparison could be moved to its own function so if the requirement changed to be case sensitive you only need to do it in one place.